hnn_core/network.py

```diff
@@ -163,53 +156,69 @@ class Network(object):
     """
 
     def __init__(self, params):
-        # set the params internally for this net
-        # better than passing it around like ...
+        # Save the parameters used to create the Network
         self.params = params
+
         # Number of time points
         # Originally used to create the empty vec for synaptic currents,
         # ensuring that they exist on this node irrespective of whether
         # or not cells of relevant type actually do
-
         self.n_times = np.arange(0., self.params['tstop'],
                                  self.params['dt']).size + 1
 
         self.n_src = 0
         self.n_of_type = {}  # numbers of sources
         self.n_cells = 0  # init self.n_cells
+        # Source list of names, first real ones only!
+        self.cellname_list = [
+            'L2_basket',
+            'L2_pyramidal',
+            'L5_basket',
+            'L5_pyramidal',
+        ]
+        self.feedname_list = []  # no feeds defined yet
+        # this will contain a list of all cell names, real and artificial
+        self._all_cell_names = []
 
-        # params of common external feeds inputs in p_common
-        # Global number of external inputs ... automatic counting
-        # makes more sense
-        # p_unique represent ext inputs that are going to go to each cell
-        self.p_common, self.p_unique = create_pext(self.params,
-                                                   self.params['tstop'])
-        self.n_common_feeds = len(self.p_common)
-        # Source list of names
-        # in particular order (cells, common, names of unique inputs)
-        self.src_list_new = self._create_src_list()
         # cell position lists, also will give counts: must be known
         # by ALL nodes
-        self.pos_dict = dict.fromkeys(self.src_list_new)
-        self.pos_dict = _create_coords(n_pyr_x=self.params['N_pyr_x'],
-                                       n_pyr_y=self.params['N_pyr_y'],
-                                       n_common_feeds=self.n_common_feeds,
-                                       p_unique_keys=self.p_unique.keys(),
-                                       zdiff=1307.4)
+        # XXX structure of pos_dict determines all downstream inferences of
+        # cell counts, real and artificial
+        self.pos_dict = dict.fromkeys(self.cellname_list)
```

Comment:
```suggestion
```

as a general rule, it will be easier to get PRs merged if there are more red lines than green :-)

Reply:
ok